File: baselib/src/main/java/com/zhoujiulong/baselib/base/ActivityManager.java
package com.zhoujiulong.baselib.base;

import android.app.Activity;

import java.util.Stack;

/**
 * 应用Activity管理类
 */
public class ActivityManager {

    private static volatile Stack<SimpleActivity> activityStack;
    private static volatile ActivityManager instance;

    private ActivityManager() {
    }

    /**
     * 获得单例对象
     *
     * @return ActivityManager
     */
    public static ActivityManager getInstance() {
        if (instance == null) {
            synchronized (ActivityManager.class) {
                if (instance == null) instance = new ActivityManager();
            }
        }
        return instance;
    }

    /**
     * 添加 Activity
     *
     * @param activity Activity
     */
    public void addActivity(SimpleActivity activity) {
        if (activityStack == null) {
            activityStack = new Stack<SimpleActivity>();
        }
        if (activity != null) activityStack.add(activity);
    }

    /**
     * 移除 Activity
     *
     * @param activity Activity
     */
    public void removeActivity(SimpleActivity activity) {
        if (activity != null && activityStack != null) {
            activityStack.remove(activity);
        }
    }

    /**
     * 结束指定Activity
     *
     * @param activity Activity
     */
    public void finishActivity(SimpleActivity activity) {
        if (activity != null) {
            if (activityStack != null) activityStack.remove(activity);
            activity.finish();
        }
    }

    /**
     * 结束指定Activity
     *
     * @param cls Activity.class
     */
    public void finishActivity(Class<?> cls) {
        if (activityStack != null) {
            for (SimpleActivity activity : activityStack) {
                if (activity != null && activity.getClass().equals(cls)) {
                    activityStack.remove(activity);
                    finishActivity(activity);
                    break;
                }
            }
        }
    }

    /**
     * 结束所有Activity
     */
    public void finishAllActivity() {
        if (activityStack != null) {
            for (Activity activity:activityStack) {
                activity.finish();
            }
            activityStack.clear();
        }
    }

    /**
     * 退出应用程序
     */
    public void appExit() {
        try {
            finishAllActivity();
            android.os.Process.killProcess(android.os.Process.myPid());
            System.exit(0);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    /**
     * 获取当前的Activity
     */
    public SimpleActivity getCurrentActivity() {
        return activityStack.lastElement();
    }

    public Stack<SimpleActivity> getActivityStack(){
        return activityStack;
    }
}

File: baselib/src/main/java/com/zhoujiulong/baselib/base/ActivityManager.kt
package com.zhoujiulong.baselib.base

import java.util.*

/**
 * 应用Activity管理类
 */
class ActivityManager private constructor() {

    companion object {

        @Volatile
        private var activityStack: Stack<SimpleActivity> = Stack()
        @Volatile
        private var instance: ActivityManager? = null

        /**
         * 获得单例对象
         *
         * @return ActivityManager
         */
        fun getInstance(): ActivityManager? {
            if (instance == null) {
                synchronized(ActivityManager::class.java) {
                    if (instance == null) instance = ActivityManager()
                }
            }
            return instance
        }
    }

    /**
     * 获取当前的Activity
     */
    val currentActivity: SimpleActivity
        get() = activityStack.lastElement()

    /**
     * 添加 Activity
     *
     * @param activity Activity
     */
    fun addActivity(activity: SimpleActivity?) {
        if (activity != null) activityStack.add(activity)
    }

    /**
     * 移除 Activity
     *
     * @param activity Activity
     */
    fun removeActivity(activity: SimpleActivity?) {
        activityStack.remove(activity)
    }

    /**
     * 结束指定Activity
     *
     * @param activity Activity
     */
    fun finishActivity(activity: SimpleActivity?) {
        if (activity != null) {
            activityStack.remove(activity)
            activity.finish()
        }
    }

    /**
     * 结束指定Activity
     *
     * @param cls Activity.class
     */
    fun finishActivity(cls: Class<*>) {
        for (activity in activityStack) {
            if (activity != null && activity.javaClass == cls) {
                activityStack.remove(activity)
                finishActivity(activity)
                break
            }
        }
    }

    /**
     * 结束所有Activity
     */
    fun finishAllActivity() {
        for (activity in activityStack) {
            activity.finish()
        }
        activityStack.clear()
    }

    /**
     * 退出应用程序
     */
    fun appExit() {
        try {
            finishAllActivity()
            android.os.Process.killProcess(android.os.Process.myPid())
            System.exit(0)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    fun getActivityStack(): Stack<SimpleActivity> {
        return activityStack
    }

}
